### PR TITLE
add ts130f tuya cover curtain

### PR DESCRIPTION
**Pull request by eyoz**
This patch add tuya cover ts130 support.
As mentioned into issue #833. The lift percentage is inverted.
Since I'm no expert at creating quirks, this fix surely needs a bit of work.

### CODE REVIEW

File: zhaquirks/tuya/ts130f.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+"""Device handler for loratap TS130F smart curtain switch."""
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.closures import WindowCovering
+from zigpy.zcl.clusters.general import Basic, Groups, OnOff, Ota, Scenes, Time
+
+from ..const import (
```
Comment (dmulcahey) [inline zhaquirks/tuya/ts130f.py:8]: please expand the import so it isn't relative

**Comment by miit86** [comment]
Hi,

I have an TS130F.

Can you explain the calibration proces. ?

Thanks in advance.

**Comment by dmulcahey** [comment]
@Adminiuga you good w/ this?

**Comment by miit86** [comment]
> > Can you explain the calibration proces. ?
> 
> Those might be hints :
> 
> * [dresden-elektronik/deconz-rest-plugin#3757 (comment)](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3757#issuecomment-741012249)
> * [dresden-elektronik/deconz-rest-plugin#3757 (comment)](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3757#issuecomment-742636504)

Hi Axel,

See below for the settings.
settings

When Read the zigbee attribute:   enum8.undefined_0x01

When Set the zigbee attribute: in log

[0xee16:1:0x0102] invalid literal for int() with base 10: 'start'
[0xee16:1:0x0102] invalid literal for int() with base 10: 'enum8.undefined_0x01'
[0xee16:1:0x0102] invalid literal for int() with base 10: 'None'
[0xee16:1:0x0102] invalid literal for int() with base 10: 'Start'

**Comment by balu79** [comment]
I also have a slightly different version of this switch, _TZ3000_egq7y6pr. How can I get this added as well (or add myself)? Anyway, this is really great, thanks a lot to everyone involved in creating this quirk!

**Comment by miit86** [comment]
IT Works. :)

I made a Quirk. and the calibrate procedure works:
I will ask to add this one as a device template.

1. close Cover
2. Set calibrate attribute to 0
3. open/up cover
4. stop cover
5. close/down cover
6. stop cover
7. set calibrate attribute to 1

**Comment by miit86** [comment]
> I also have a slightly different version of this switch, _TZ3000_egq7y6pr. How can I get this added as well (or add myself)? Anyway, this is really great, thanks a lot to everyone involved in creating this quirk!

the Steps:

1. open home assistant in WINSCP (to copy or edit files)
2. navigate to: config/deps/lib/python3.8/site-packages/zhaquirks/tuya
3. copy the TS130F.py to this folder.
4. edit/add this line: MODELS_INFO: [("_TZ3000_vd43bbfq", "TS130F"), ("_TZ3000_egq7y6pr", "TS130F") ],
5. reboot HA
6. check if the quirk loads.

if it does not work can you post your device template?

**Comment by balu79** [comment]
> > I also have a slightly different version of this switch, _TZ3000_egq7y6pr. How can I get this added as well (or add myself)? Anyway, this is really great, thanks a lot to everyone involved in creating this quirk!
> 
> the Steps:
> 
> 1. open home assistant in WINSCP (to copy or edit files)
> 2. navigate to: config/deps/lib/python3.8/site-packages/zhaquirks/tuya
> 3. copy the TS130F.py to this folder.
> 4. edit/add this line: MODELS_INFO: [("_TZ3000_vd43bbfq", "TS130F"), ("_TZ3000_egq7y6pr", "TS130F") ],
> 5. reboot HA
> 6. check if the quirk loads.
> 
> if it does not work can you post your device template?

Yes, i can confirm it works, calibration, inverted position and other aspects, all fine

**Pull request merged**